python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -157,6 +167,55 @@ def point_indices(self):
                 "Polygons to return point indices."
             )
 
+    @property
+    def sizes(self):
+        """Returns the number of points of each geometry in the GeoSeries."
+
+        Returns
+        -------
+        sizes : cudf.Series
+            The size of each geometry in the GeoSeries.
+
+        Notes
+        -----
+        The size of a geometry is the number of points it contains.
+        The size of a polygon is the number of points in its exterior ring
+        plus the number of points in its interior rings.
+        The size of a multipolygon is the sum of all its polygons.
+        The size of a linestring is the number of points in its single line.
+        The size of a multilinestring is the sum of all its linestrings.
+        The size of a multipoint is the number of points in its single point.
+        The size of a point is 1.
+        """
+        if contains_only_polygons(self):
```

Comment:
The reason for this logic is because `sizes` returns the size of each row of the `GeoSeries`. For example 
```
x = cuspatial.GeoSeries([
    LineString([(0, 0), (1, 1)]),
    LineString([(0, 0), (1, 1), (2, 2)]),
    LineString([(0, 0), (1, 1), (2, 2), (3, 3)]),
])
print(x.sizes)
```
```
0    2
1    3
2    4
dtype: int32
```
The .xy array in this case contains all of the points for all of the features in the GeoSeries:
```
x.lines.xy
```
```
0     0.0
1     0.0
2     1.0
3     1.0
4     0.0
5     0.0
6     1.0
7     1.0
8     2.0
9     2.0
10    0.0
11    0.0
12    1.0
13    1.0
14    2.0
15    2.0
16    3.0
17    3.0
dtype: float64
```
But has no information about the lengths of the individual geometries.

Reply:
`sum(x.sizes) == 9` and `len(x.lines.xy) / 2 == 9`. So why doesn't the latter serve the purpose?

Reply:
For example, the sizes are used for each row of the `GeoSeries` to determine if the `.contains` (boundary inclusive) predicate has been met:

```
return contains + intersects >= rhs.sizes
```
Where contains is a `cudf.Series` of the number of points that satisfied `.contains_properly` for each row in the GeoSeries, and intersects is a `cudf.Series` of the number of points that satisfied `.intersects` for those same rows in the GeoSeries. So if I had a GeoSeries of a length-three polygon and a length 4 polygon, sizes would be ```
0    3
1    4
dtype: int32
```
and in the cast of a self-contains test contains would be
```
0    0
1    0
dtype: int32
```
and intersects would be
```
0    3
0    4
dtype: int32
```
so the predicate result of `.contains` is
`contains_properly + intersects >= rhs.sizes ==
```
0    True
1    True
dtype: bool
```

Does that properly answer your question? I'm not using the sum of the sizes in the `GeoSeries`, but each individually.